tests/test_field.py

```diff
@@ -25,3 +25,12 @@ def test_invalid_operations_on_fields():
     for op in operators:
         with pytest.raises(FieldsNotIdentical):
             op(field1(2), field2(3))
+
+
+def test_sqrt():
```

Comment:
I'd rather see sqrt on a few random tests (say 50? should still be fast) than just one hardcoded one

Reply:
Will do.

Reply:
We should write test cases to cover both `if` and `else` cases and then a few random ones.